src/IntegrationUtils.ts

```diff
@@ -0,0 +1,129 @@
+			'Secret - Apache Camel K Integration with Kubernetes Secret'], 
+		{
+			placeHolder: 'Select the type of Apache Camel K Integration'
+		}).then( async (choice) => {
+			let selectedConfigMap = undefined;
+			let selectedSecret = undefined;
+
+			switch (choice) {
+				case 'ConfigMap - Apache Camel K Integration with Kubernetes ConfigMap':
+					await getConfigMaps().then( async (configMaps) => {
+						selectedConfigMap = await vscode.window.showQuickPick(configMaps, {
+							placeHolder: 'Select an available Kubernetes ConfigMap or ESC to cancel'
+						});
+					});
+					break;
+				case 'Secret - Apache Camel K Integration with Kubernetes Secret':
+					await getSecrets().then( async (secrets) => {
+						selectedSecret = await vscode.window.showQuickPick(secrets, {
+							placeHolder: 'Select an available Kubernetes Secret or ESC to cancel'
+						});
+					});
+					break;
+				case 'Basic - Apache Camel K Integration (no ConfigMap or Secret)':
+					// do nothing with config-map or secret
+					break;
+			}
+
+			if (!choice) {
+				reject(new Error('No integration selection made.'));
+				return;
+			}
+	
+			await createNewIntegration(context, selectedConfigMap, selectedSecret)
+				.then( success => {
+					if (!success) {
+						reject(false);
+					}
+					resolve(true);
+				})
+				.catch(err => {
+					reject(err);
+				});
+			});
+		});
+}
+		
+// use command-line "kamel" utility to start a new integration
+function createNewIntegration(integrationFileUri: vscode.Uri, configmap? : string, secret? : string): Promise<boolean> {
+	return new Promise( async (resolve, reject) => {
+		let filename = integrationFileUri.fsPath;
+		let foldername = path.dirname(filename);
+		let absoluteRoot = path.parse(filename).base;
+		let rootName = absoluteRoot.split('.', 1)[0];
+		let integrationName = utils.toKebabCase(rootName);
+		utils.shareMessage(extension.mainOutputChannel, `Deploying file ${absoluteRoot} as integration ${integrationName}`);
+		await extension.removeOutputChannelForIntegrationViaKubectl(integrationName)
+			.catch( (error) => { 
+				// this is not a hard stop, it just means there was no output channel to close
+				console.error(error); 
+			});
+
+		let commandString = 'kamel run';
+		if (configmap && configmap.trim().length > 0) {
+			commandString += ` --configmap=${configmap}`;
+		}
+		if (secret && secret.trim().length > 0) {
+			commandString += ` --secret=${secret}`;
+		}
+		commandString += ' "' + absoluteRoot + '"';
+		console.log(`commandString = ${commandString}`);
+		let runKubectl = child_process.exec(commandString, { cwd : foldername});
+		runKubectl.stdout.on('data', function (data) {
+			resolve(true);
+			return;
+		});
+		runKubectl.stderr.on('data', function (data) {
+			utils.shareMessage(extension.mainOutputChannel, `Error deploying ${integrationName}: ${data}`);
+			reject(false);
+			return;
+		});
+	});
+}		
+
+export async function isCamelKAvailable(): Promise<any> {
```

Comment:
I would recommend to replace any with boolean because you always return a boolean value.